src/main/java/com/atlassian/db/replica/internal/aurora/AuroraClusterDiscovery.java

```diff
@@ -11,17 +12,21 @@
 import static java.util.stream.Collectors.toList;
 
 public final class AuroraClusterDiscovery {
-    private final AuroraConnectionDetails auroraConnectionDetails;
+    private final ConnectionPoolPerUrlProvider connectionPoolPerUrlProvider;
 
-    public AuroraClusterDiscovery(AuroraConnectionDetails auroraConnectionDetails) {
-        this.auroraConnectionDetails = auroraConnectionDetails;
+    public AuroraClusterDiscovery(ConnectionPoolPerUrlProvider connectionPoolPerUrlProvider) {
```

Comment:
I think its better to extract `Supplier<Connection> getConnectionSupplier` instead of 
`ConnectionPoolPerUrlProvider`
Why ? We shouldn't play with `connectionPool` in `AuroraClusterDiscovery`  this is to greedy IMHO. 
AuroraClusterDiscovery just want to get connection for URL
`ConnectionUrlProvider`

Reply:
It will give more control on product side around connectionPool and connection, and I think this is what we want to achieve yeah ?

Reply:
another advantage is 1 less SPI to maintain `com.atlassian.db.replica.spi.ConnectionPool`